src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -188,7 +210,8 @@ static void replace_multi_inst_dep(char *src) {
                 } else {
                     service_name = strdup(word);
                     service_name = strtok_r(service_name, ".", &save_ptr2);
-                    type = strtok_r(NULL, " ", &save_ptr2);
+                    type = strtok_r(NULL, "", &save_ptr2);
+                    type = strtok(type, "\n");
```

Comment:
Why are these changes needed?

Reply:
This is to remove extra end of line from the string.
Later below "snprintf(buf, MAX_BUF_SIZE, "%s=%s@%d.%s\n"," adds the needed end of line.
This is to prevent extra empty lines in unit files.

Reply:
Couldn't these two lines be simplified to

```
type = strtok_r(NULL, "\n", &save_ptr2);
```

Reply:
The original string may contain multiple service strings separated by white space:
After=unit1.service unit2.service

I was thinking the first statement was doing a split on white space and added Second to remove "\n" if any.
But that whitespace split happens before in while check itself so yes, we don't need two statements. Just one with "\n" is good enough. 
I will make the change, test and upload.

Reply:
@theasianpianist I have committed the change. You may have to approve again. Thanks!
@jleveque Please approve if you don't have any more comment. Thanks